app/forms/provider_interface/offer_wizard.rb

```diff
@@ -38,14 +45,55 @@ def valid_for_current_step?
       valid?(current_step.to_sym)
     end
 
-    def next_step
-      index = STEPS[decision.to_sym].index(current_step.to_sym)
+    def next_step(step = current_step)
+      index = STEPS[decision.to_sym].index(step.to_sym)
+      return unless index
+
+      current_step = STEPS[decision.to_sym][index + 1]
+
+      return save_and_go_to_next_step(current_step) if current_step.eql?(:providers) && available_providers.one?
+      return save_and_go_to_next_step(current_step) if current_step.eql?(:courses) && available_courses.one?
+      return save_and_go_to_next_step(current_step) if current_step.eql?(:study_modes) && available_study_modes.one?
+      return save_and_go_to_next_step(current_step) if current_step.eql?(:locations) && available_course_options.one?
```

Comment:
I'm running into an issue locally where a course does not have course options and so we're getting an empty locations page as this is checking if its one and not if any exists, I checked on production and we do have some of those, is it safer to check if any exist as well? Also I'm not sure if this is actually a real issue as those courses will probably not be valid when we plug in the real query service?
![Screenshot 2021-03-18 at 10 35 16](https://user-images.githubusercontent.com/2732945/111597052-7de76580-87d6-11eb-8d92-ff1914ca42b4.png)

Reply:
We raised this to Michael and expect the service retrieving the available data to take this into account and not return any options that will result in a dead-end.